src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1122,54 +1122,9 @@ public function getPowType(Expr $left, Expr $right, callable $getTypeCallback):
 		$leftType = $getTypeCallback($left);
 		$rightType = $getTypeCallback($right);
 
-		if ($leftType instanceof MixedType || $rightType instanceof MixedType) {
-			return new BenevolentUnionType([
-				new FloatType(),
-				new IntegerType(),
-			]);
-		}
-
-		$object = new ObjectWithoutClassType();
-		if (
-			!$leftType instanceof NeverType &&
-			!$rightType instanceof NeverType &&
-			(
-				!$object->isSuperTypeOf($leftType)->no()
-				|| !$object->isSuperTypeOf($rightType)->no()
-			)
-		) {
-			return TypeCombinator::union($leftType, $rightType);
-		}
-
-		$leftTypes = TypeUtils::getConstantScalars($leftType);
-		$rightTypes = TypeUtils::getConstantScalars($rightType);
-		$leftTypesCount = count($leftTypes);
-		$rightTypesCount = count($rightTypes);
-		if ($leftTypesCount > 0 && $rightTypesCount > 0) {
-			$resultTypes = [];
-			$generalize = $leftTypesCount * $rightTypesCount > self::CALCULATE_SCALARS_LIMIT;
-			foreach ($leftTypes as $leftTypeInner) {
-				foreach ($rightTypes as $rightTypeInner) {
-					$leftNumberType = $leftTypeInner->toNumber();
-					$rightNumberType = $rightTypeInner->toNumber();
-
-					if ($leftNumberType instanceof ErrorType || $rightNumberType instanceof ErrorType) {
-						return new ErrorType();
-					}
-
-					if (!$leftNumberType instanceof ConstantScalarType || !$rightNumberType instanceof ConstantScalarType) {
-						throw new ShouldNotHappenException();
-					}
-
-					$resultType = $this->getTypeFromValue($leftNumberType->getValue() ** $rightNumberType->getValue());
-					if ($generalize) {
-						$resultType = $resultType->generalize(GeneralizePrecision::lessSpecific());
-					}
-					$resultTypes[] = $resultType;
-				}
-			}
-
-			return TypeCombinator::union(...$resultTypes);
+		$exponentiatedTyped = $leftType->exponentiate($rightType);
+		if (!$exponentiatedTyped instanceof ErrorType) {
+			return $exponentiatedTyped;
 		}
 
 		return $this->resolveCommonMath(new BinaryOp\Pow($left, $right), $leftType, $rightType);
```

Comment:
I kept this pass thru `resolveCommonMath` because otherwise we would loose support for operator-type specifiying extensions (even though atm, there is no test-coverage for using it with `pow`).

Reply:
test added with https://github.com/phpstan/phpstan-src/pull/1933